src/Sarif.Multitool/Rules/SARIF1009.IndexPropertiesMustBeConsistentWithArrays.cs

```diff
@@ -163,11 +164,28 @@ protected override void Analyze(WebResponse webResponse, string webResponsePoint
             string propertyName,
             string arrayName)
         {
+            if (index == -1)
+            {
+                return;
+            }
+
+            if (container == null)
+            {
+                LogResult(
+                    jsonPointer,
+                    nameof(RuleResources.SARIF1009_IndexPropertiesMustBeConsistentWithArrays_Error_TargetArrayMustExist_Text),
+                    objectName,
+                    propertyName,
+                    index.ToInvariantString(),
+                    arrayName);
+                return;
+            }
```

Comment:
this is the worthwhile change in this PR (other things are renames/housekeeping).

we now have two messages depending on whether the collection was empty or did not have enough elements. The test case output is accordingly modified :).